File: Sorting/Bubble Sort/bubble_sort.c
/* Bubble Sorting Algorithm in C */
// =================================================
#include<stdio.h>

void bubble_sort(int arr[],int n) {
    int i, j, temp;
    for (i = 0; i < n; i++) {
        for (j = 0; j < n-i-1; j++) {
            if (arr[j] > arr[j+1]) {
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

void printarr(int arr[], int n){
    printf("\n");
    for(int i=0; i<n; i++){
        printf(" %d ",arr[i]);
    }
    printf("\n");
}

void main(){
    int arr[5] = {5,1,3,2,4}, n = 5;

    printf("\nList before sorting");
    printarr(arr, n);

    bubble_sort(arr, n);

    printf("\nList after sorting");
    printarr(arr, n);
}
// ================================================
/* by Abel Roy */

File: Sorting/Bubble Sort/bubble_sort.py
# Bubble Sorting Algorithm in Python
# =================================================
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range (0,n-i-1):
            if arr[j]>arr[j+1]:
                arr[j],arr[j+1]=arr[j+1],arr[j]
    return arr

def main():
    arr = [8,10,2,6,9]

    print("List before sorting: ",arr)
    arr = bubble_sort(arr)
    print("List after sorting: ",arr)

main()
# ================================================
# By Abel Roy #
